packages/core/src/stateUtils.ts

```diff
@@ -1184,17 +1202,20 @@ export function resolveTransition<
           SpecialTargets.Internal)
   );
 
-  let children = currentState ? currentState.children : [];
+  let children = currentState ? currentState.children : {};
   for (const action of resolvedActions) {
     if (action.type === actionTypes.start) {
-      children.push(createInvocableActor((action as any).actor));
+      const actorDef = createInvocableActor(
+        (action as InvokeActionObject<TContext, TEvent>).def
+      );
+      children[actorDef.id] = actorDef;
+      // TODO: warn of collisions
     } else if (action.type === actionTypes.stop) {
-      children = children.filter(childActor => {
-        return (
-          childActor.id !==
-          (action as ActivityActionObject<TContext, TEvent>).actor.id
-        );
-      });
+      const id = (action as InvokeActionObject<TContext, TEvent>).def.id;
+
+      children = { ...children };
+      delete children[id];
```

Comment:
I wonder if we really need to represent them at `state.children` right away as "null actors". Why exactly this is important? It would be better to get back to this problem~ once we get to making the transition a microstep, I have a feeling that it might impact this thing.